ioxd_common/src/http/error.rs

```diff
@@ -109,18 +121,18 @@ impl HttpApiError {
         Self {
             code: code.into(),
             msg: msg.into(),
+            line: None,
         }
     }
 
+    /// Add body to error.
+    pub fn with_line(self, line: Option<usize>) -> Self {
+        Self { line, ..self }
+    }
+
     /// Generate response body for this error.
     fn body(&self) -> Body {
-        let json = serde_json::json!({
-            "code": self.code.as_text().to_string(),
-            "message": self.msg.clone(),
-        })
-        .to_string();
-
-        Body::from(json)
+        Body::from(serde_json::json!(self).to_string())
```

Comment:
I believe this would normally use `serde_json::to_string()`:

```suggestion
        Body::from(serde_json::to_string(&self).expect("must serialise to json"))
```